cmd/src/server.rs

```diff
@@ -107,9 +110,14 @@ pub fn run_tikv(config: TiKvConfig) {
         ($ER: ty) => {{
             let enable_io_snoop = config.enable_io_snoop;
             let mut tikv = TiKVServer::<$ER>::init(config);
-            if enable_io_snoop {
+            let recorder = if enable_io_snoop {
                 tikv.init_io_snooper();
-            }
+                None
+            } else {
+                let recorder = Arc::new(BytesRecorder::new());
```

Comment:
can we make the iosnoop to be a kind of BytesRecorder?

Reply:
Since iosnoop doesn't interact with limiter the same way as bytes recorder, I don't think so. It might be better to extract the global IOContext out. I'll leave it to another PR where we actually use IOContext.

Reply:
Seems the difference is the `add()`  of `BytesRecorder`. For iosnoop, we can just do nothing in `add()` and use `IOContext` to get the difference for `fetch()`

Reply:
Latency isn't included in bytes recorder, and doing that means spreading the optional compilation directive all over the place.

Reply:
Latency is just for metrics, you don't consider it in BytesRecorder. And seems optional compilation can be bounded by file_system providing a uniform interface like `create_bytes_recorder(some_config)`.

Reply:
Got it, i thought latency is stored in IOContext too.